// @gemwallet/api/_constants/network/network.constant.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.NETWORK = exports.NetworkServer = exports.Network = void 0;
var Network;
(function (Network) {
    Network["MAINNET"] = "Mainnet";
    Network["TESTNET"] = "Testnet";
    Network["DEVNET"] = "Devnet";
    Network["AMM_DEVNET"] = "AMM-Devnet";
    Network["CUSTOM"] = "Custom";
})(Network = exports.Network || (exports.Network = {}));
var NetworkServer;
(function (NetworkServer) {
    NetworkServer["MAINNET"] = "wss://xrplcluster.com";
    NetworkServer["TESTNET"] = "wss://s.altnet.rippletest.net:51233";
    NetworkServer["DEVNET"] = "wss://s.devnet.rippletest.net:51233";
    NetworkServer["AMM_DEVNET"] = "wss://amm.devnet.rippletest.net:51233";
})(NetworkServer = exports.NetworkServer || (exports.NetworkServer = {}));
exports.NETWORK = (_a = {},
    _a[Network.MAINNET] = {
        name: Network.MAINNET,
        server: NetworkServer.MAINNET,
        description: 'Main network using the production version of the XRP Ledger.'
    },
    _a[Network.TESTNET] = {
        name: Network.TESTNET,
        server: NetworkServer.TESTNET,
        description: 'Acts as a testing network, without impacting production users and risking real money.'
    },
    _a[Network.DEVNET] = {
        name: Network.DEVNET,
        server: NetworkServer.DEVNET,
        description: 'A preview of upcoming features, where unstable changes are tested out.'
    },
    _a[Network.AMM_DEVNET] = {
        name: Network.AMM_DEVNET,
        server: NetworkServer.AMM_DEVNET,
        description: 'XLS-30d Automated Market Makers preview network.'
    },
    _a[Network.CUSTOM] = {
        name: Network.CUSTOM,
        server: '',
        description: 'Custom network'
    },
    _a);
